MekHQ/src/mekhq/campaign/mission/Scenario.java

```diff
@@ -736,16 +736,12 @@ public boolean isTraitor(Entity en, Campaign c) {
         }
         // also make sure that the crew's external id does not match a traitor in
         // case of ejected pilots
-        if ((null != en.getCrew()) && !"-1".equals(en.getCrew().getExternalIdAsString()) &&
-                isTraitor(UUID.fromString(en.getCrew().getExternalIdAsString()))) {
-            return true;
-        }
-        return false;
+        return (null != en.getCrew()) && !"-1".equals(en.getCrew().getExternalIdAsString()) && isTraitor(UUID.fromString(en.getCrew().getExternalIdAsString()));
```

Comment:
split this into individual variables, test each one independently and then do the final check which says if they are ... well, whatever this is supposed to say.

Reply:
I'll need to double check this, but I think it might actually be pre-existing code or potentially a bad merge. I can't remember why I made this change, making me think it might not have been intentional. I'll review and correct accordingly. :)

Reply:
this looks like just an aggregation/simplification of a condition check, the original was also very bad.

the "best" here would be to do something like this:

```java

var crewIsPresent = en.getCrew() != null
if (!crewIsPresent) {
  return false;
}
var crewExists = en.getCrew.getExternalIdAsString() != "-1";
var crewIsTraitor = isTraitor(UUID.fromString(en.getCrew().getExternalId()));
return crewExists && !crewIsTraitor;
```

PS.: I dont know if those names are correct or if thats the function of the method, this is just an example.